h2/src/main/org/h2/util/SourceCompiler.java

```diff
@@ -168,6 +179,32 @@ private static boolean isGroovySource(String source) {
         return source.startsWith("//groovy") || source.startsWith("@groovy");
     }
 
+    private static boolean isJavascriptSource(String source) {
+        return source.startsWith("//javascript");
+    }
+
+    // whether the passed source should be compiled using javax.script.ScriptEngineManager
+    private static boolean isJavaxScriptSource(String source) {
+        return isJavascriptSource(source);
+    }
+
+    public CompiledScript getCompiledScript(String packageAndClassName) throws ScriptException {
```

Comment:
this can be private, and just call it getCompiledJavascript, since we don't support anything else yet